### PR TITLE
Add a dispatch version number to netcdf_meta.h and libnetcdf.settings, in case we decide to change dispatch table in future

### LINKED ISSUE #1469: add a dispatch table version number to netcdf_meta.h and libnetcdf.settings
This is part of #1056 

Should there be some version number in the dispatch table?

How would that work? Let's say we call the current dispatch table (after merging outstanding PRs) version 1.

Then, some package, like PIO, uses the dispatch layer and everyone is happy.

Then, we decide to change the dispatch table - we add a function. So we bump the version.

How does the version number help? PIO will still have to change its struct to include the new function. So there will be no way to use the version to keep using the old dispatch table, without the new funciton.

So what is the point of a version?

### PR DESCRIPTION
**Pull request by edhartnett**
Fixes #1469 

Adding a version number for the dispatch table, in case we decide to change it at some point.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1971,6 +1971,7 @@ is_enabled(JNA HAS_JNA)
 is_enabled(ENABLE_ZERO_LENGTH_COORD_BOUND RELAX_COORD_BOUND)
 is_enabled(USE_CDF5 HAS_CDF5)
 is_enabled(ENABLE_ERANGE_FILL HAS_ERANGE_FILL)
+is_enabled(DISPATCH_VERSION 1)
```
Comment (edhartnett) [inline CMakeLists.txt]: @WardF is this the correct way to do this in CMake?

Comment (WardF) [reply CMakeLists.txt]: @edhartnett `is_enabled` is a macro created to map a 0/1, 'true/false' to an autotools-style 'yes/no'.  So in the example of `is_enabled(USE_CDF5 HAS_CDF5)`, `is_enabled` evaluates the value of `USE_CDF5` and assigns `HAS_CDF5` a value of either `yes` or `no`.  I believe these are then used when generating `nc-config` and `libnetcdf.settings`.

So, in short, I don't think `is_enabled(DISPATCH_VERSION 1)` does what you want it to do.

Comment (edhartnett) [reply CMakeLists.txt]: ...so what should I do instead? ;-)

Comment (WardF) [reply CMakeLists.txt]: At an appropriate place in `CMakeLists.txt` (I would suggest near the top, where we define the netcdf version and SO version), add a `SET(NC_DISPATCH_VERSION 1)` statement.  This should propagate throughout when cmake generates files that reference it :).

Comment (edhartnett) [reply CMakeLists.txt]: Will do...

**Pull request merged**